src/js/rtc_peer_connection_factory.js

```diff
@@ -0,0 +1,161 @@
+    _pcOnsignalingstatechangeEventHandler(event) {// eslint-disable-line no-unused-vars
+        var self = this;
+        //Chrome supported connectionState in RtcPeerConnection from version 72. Before that signalingState was used to determine the status of the RtcPeerConnection
+        if (self._pc.connectionState === "closed" || self._pc.signalingState === "closed") {
+            self._pc = null;
+            if (!self._isConnectionIdle && !self._isQuotaBreached) {
+                self._requestPeerConnections();
+            }
+        }
+    }
+
+    //This method will listen the signaling state change of backup pc and refresh the backup pc except connection was idle or user has breached the quota limit
+    _backupPcOnsignalingstatechangeEventHandler(event) {// eslint-disable-line no-unused-vars
+        var self = this;
+        //Chrome supported connectionState in RtcPeerConnection from version 72. Before that signalingState was used to determine the status of the RtcPeerConnection
+        if (self._backupPc.connectionState === "closed" || self._pc.signalingState === "closed") {
+            self._backupPc = null;
+            if (!self._isConnectionIdle && !self._isQuotaBreached) {
+                self._requestPeerConnections();
+            }
+        }
+    }
+
+    _requestPeerConnections() {
+        var self = this;
+        if (self._browserSupported) {
+            self._requestIceAccess().then(function (response) {
+                    self._initializeRtcPeerConnections(response.softphoneTransport.softphoneMediaConnections);
+                },
+                // eslint-disable-next-line no-unused-vars
+                function (reason) {
+                });
+        }
+    }
+
+    //This method will initialize the pc and backup. This method will also set the timeout for managing idle connections
+    _initializeRtcPeerConnections(iceServers) {
+        var self = this;
+        if (self._pc == null) {
+            self._pc = self._createRtcPeerConnection(iceServers, self._pcOnsignalingstatechangeEventHandler);
+        }
+        if (self._backupPc == null) {
+            self._backupPc = self._createRtcPeerConnection(iceServers, self._backupPcOnsignalingstatechangeEventHandler);
+        }
+        self._idleRtcPeerConnectionsTimerId = setTimeout(hitch(self, self._clearIdleRtcPeerConnections), RTC_PEER_CONNECTION_IDLE_TIMEOUT_MS);
+    }
+
+    _createRtcPeerConnection(iceServers, signalingstatechangeEventHandler) {
+        var self = this;
+        RTC_PEER_CONNECTION_CONFIG.iceServers = iceServers;
+        RTC_PEER_CONNECTION_CONFIG.iceCandidatePoolSize = DEFAULT_ICE_CANDIDATE_POOL_SIZE;
+        var pc = new RTCPeerConnection(RTC_PEER_CONNECTION_CONFIG, RTC_PEER_CONNECTION_OPTIONAL_CONFIG);
+        if (signalingstatechangeEventHandler) {
+            pc.onsignalingstatechange = hitch(self, signalingstatechangeEventHandler);
```

Comment:
preventing user (or even ourselves) from overriding this event handler is tricky.